src/pages/ProductPage.js

```diff
@@ -21,30 +22,36 @@ const ProductPage = () => {
   let itemId = useParams().id;
   const [data, setData] = useState({ products: [] });
   const [isFetching, setIsFetching] = useState(false);
+
+  const [message, setMessage] = useState('');
   useEffect(() => {
     const fetchData = async () => {
       try {
         setIsFetching(true);
-        setData({ products: data.products, isFetching: true });
+        setMessage('Loading...');
+        setData({ products: data.products });
```

Comment:
This line isn't necessary since you are just setting data to what it is already.

Reply:
noted:D